src/test/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/integration/PlanControllerTest.kt

```diff
@@ -81,25 +88,46 @@ class PlanControllerTest : IntegrationTestBase() {
   @Nested
   @DisplayName("createNewGoal")
   inner class CreateNewGoal {
-    lateinit var goalRequestBody: GoalEntity
+    lateinit var goalRequestBody: Goal
 
     @BeforeEach
     fun setup() {
-      goalRequestBody = GoalEntity(
+      goalRequestBody = Goal(
         title = "abc",
-        areaOfNeed = "xzv",
+        areaOfNeed = "Accommodation",
         targetDate = LocalDateTime.now().toString(),
-        goalOrder = 1,
       )
     }
 
     @Test
-    fun `should return created when creating goal`() {
-      webTestClient.post().uri("/plans/$planUuid/goals").header("Content-Type", "application/json")
+    fun `should return created when creating goal with no related areas of need`() {
+      val goalEntity: GoalEntity = webTestClient.post().uri("/plans/$planUuid/goals").header("Content-Type", "application/json")
         .headers(setAuthorisation(user = "Tom C", roles = listOf("ROLE_RISK_INTEGRATIONS_RO")))
         .bodyValue(goalRequestBody)
         .exchange()
         .expectStatus().isCreated
+        .expectBody<GoalEntity>()
+        .returnResult().responseBody
```

Comment:
For this, and the below test, the call to responseBody is @Nullable annotated, so the return type should be `GoalEntity?`

Reply:
By the compiler, yes, but it then makes the assertion below harder to read (IMO), and if this return is null the test will fail so I rated my subjective legibility ranking higher than the compiler's opinion. Would you like me to change it anyway?